### PR TITLE
Create `link` variant of `Button`

### LINKED ISSUE #116: Create `link` variant of `Button`
It proves necessary there should be a component which has semantics of `<button>` but appears as link. Creating `link` variant of `Button` seems to be the best way. Styles should be shared with link styles.

### PR DESCRIPTION
**Pull request by adamkudrna**
Closes #116 

image
image
image

### CODE REVIEW

File: src/lib/components/ui/Button/Button.jsx [hunk 1]
```diff
@@ -5,48 +5,79 @@ import withForwardedRef from '../withForwardedRef';
 import styles from './Button.scss';
 
 export const Button = (props) => {
```
Comment (mbohal) [inline src/lib/components/ui/Button/Button.jsx:7]: I would write it as:
```
export const Button = ({
    afterLabel,
    beforeLabel,
    block,
    clickHandler,
    disabled,
    endCorner,
    forwardedRef,
    grouped,
    id,
    label,
    labelVisibility,
    loadingIcon,
    priority,
    size,
    startCorner,
    type,
    variant,
  }) => {
```

Comment (adamkudrna) [reply src/lib/components/ui/Button/Button.jsx:7]: I see, but… really? That's yet another approach, not yet used anywhere in the library…

Comment (mbohal) [reply src/lib/components/ui/Button/Button.jsx:7]: I think the suggested approach is more concise. I don't think it matters, that it it not used anywhere else - its not a big thing, just syntactic sugar.

Comment (mbohal) [reply src/lib/components/ui/Button/Button.jsx:7]: It is not possible due to `transferProps` functionality.

**Pull request merged**